### PR TITLE
Added labels to the agones.allocator

### LINKED ISSUE #3093: Helm option: agones.allocator.labels
**Is your feature request related to a problem? Please describe.**
We have `agones.allocator.annotations` as helm options for the Allocator, but not `agones.allocator.labels` .

This is useful when using a cross cluster service mesh, such as Istio/Anthos Service Mesh and want to be able to use the label to route requests via a single domain.

In the [global scale demo](https://github.com/googleforgames/global-multiplayer-demo) we had to use kustomize to patch the yaml config generated from helm. It would be nice not to have to do that.

See: https://github.com/googleforgames/global-multiplayer-demo/blob/main/infrastructure/files/agones/agones-install.yaml.tpl

(Yes, that Skaffold config generated by Terraform, that will get passed to Cloud Deploy).

**Describe the solution you'd like**

 `agones.allocator.labels`  that adds labels to the Deployment of the allocator.

**Describe alternatives you've considered**

Keep using Kustomize, but it's an extra unnecessary step.

**Additional context**

Using a cross cluster service mesh is actually a _really_ nice way to do multi-cluster allocation!

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #3093

**Special notes for your reviewer**:

### CODE REVIEW

File: install/helm/agones/values.yaml [hunk 1]
```diff
@@ -35,7 +35,7 @@ agones:
   serviceaccount:
     allocator:
       name: agones-allocator
-      annotations: {}
+      labels: {}
```
Comment (markmandel) [inline install/helm/agones/values.yaml:38]: We need both annotations and labels please.

Also, we'll need updates to the documentation as well.

Comment (Kalaiselvi84) [reply install/helm/agones/values.yaml:39]: I included both..

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 2]
```diff
@@ -201,6 +201,7 @@ The following tables lists the configurable parameters of the Agones chart and t
...
 | `agones.allocator.affinity`                              | Allocator [affinity][affinity] settings for pod assignment                                                                                                                                                              | `{}`                               |
 | `agones.allocator.annotations`                           | [Annotations][annotations] added to the Agones allocator pods                                                                                                                                                           | `{}`                               |
+| `agones.allocator.labels`                           | [Labels][labels] added to the Agones allocator pods                                                                                                                                                           | `{}`                               |
```
Comment (markmandel) [inline site/content/en/docs/Installation/Install Agones/helm.md:204]: This table will need to added to the feature shortcoded table at the bottom so they don't show up in documentation when merged into `main` (docs get updated immeadiately)
https://agones.dev/site/docs/contribute/#within-a-page

https://github.com/googleforgames/agones/pull/2961 is a good example

**Review by markmandel — COMMENTED** [review]
Small docs change, but otherwise good to go.

File: site/content/en/docs/Installation/Install Agones/helm.md (hunk 2 above)
Comment (Kalaiselvi84) [reply site/content/en/docs/Installation/Install Agones/helm.md]: I've updated the helm file, please review and let me know if there's anything that needs further attention..

File: install/helm/agones/values.yaml [hunk 3]
```diff
@@ -36,6 +36,7 @@ agones:
     allocator:
       name: agones-allocator
       annotations: {}
+      labels: {}
```
Comment (markmandel) [inline install/helm/agones/values.yaml:39]: I just realised something! This value hasn't been applied to the actual service Deployment! We will need to do that as well.

https://github.com/googleforgames/agones/blob/main/install/helm/agones/templates/service/allocation.yaml

**Review by markmandel — COMMENTED** [review]
Just realised this was missing a step.

Also hope you don't mind but I edited the description to add the "closes" to the issue so it would actually close the issue.

File: install/helm/agones/values.yaml (hunk 3 above)
Comment (Kalaiselvi84) [reply install/helm/agones/values.yaml:39]: I have added labels in this allocation.yaml file, please share your suggestions

File: install/helm/agones/templates/service/allocation.yaml [hunk 4]
```diff
@@ -160,6 +160,9 @@ spec:
 {{- end }}
 {{- if .Values.agones.allocator.annotations }}
 {{- toYaml .Values.agones.allocator.annotations | nindent 8 }}
+{{- end }}
+{{- if .Values.agones.allocator.labels }}
```
Comment (markmandel) [inline install/helm/agones/templates/service/allocation.yaml:164]: Looks good - just checking if you had a chance to do any manual testing on this? (I don't think there's any better way to actually test this).

Easiest way might be to use `make install` and pass in some extra `HELM_ARGS` and make sure the Deployment labels are good.

Comment (Kalaiselvi84) [reply install/helm/agones/templates/service/allocation.yaml:164]: I have validated the labels manually per your guidance

Comment (markmandel) [reply install/helm/agones/templates/service/allocation.yaml:164]: Thanks for the validation - but I just realised after looking at it, that these values are showing up in the `annotations:` section.

We should move this code up to above line 152, so it sits under `labels:` - then it should be good to go.

Comment (Kalaiselvi84) [reply install/helm/agones/templates/service/allocation.yaml]: The values have been moved to the labels section

**Review by markmandel — APPROVED** [review]
Looks good!

**Review by markmandel — APPROVED** [review]
Looks good!

**Pull request merged**